app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailContent.kt

```diff
@@ -581,19 +578,9 @@ fun BelowHeader(galleryDetail: GalleryDetail) {
             text = torrentText,
             onClick = {
                 launchIO {
-                    if (galleryDetail.torrentCount > 0) {
-                        val granted = isAtLeastQ || requestPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE)
-                        if (granted) {
-                            runSuspendCatching {
-                                showTorrentDialog()
-                            }.onFailure {
-                                showSnackbar(it.displayString())
-                            }
-                        } else {
-                            showSnackbar(permissionDenied)
-                        }
-                    } else {
-                        showSnackbar(noTorrents)
+                    when {
+                        galleryDetail.torrentCount <= 0 -> showSnackbar(noTorrents)
+                        else -> runSwallowingWithUI { showTorrentDialog() }
```

Comment:
nit: Is it necessary to use `when` expression when there are only two branches?

Reply:
As content body is not complicate, when expr looks a bit tidy than if else, anyway it does no harm.